models/userModel.js

```diff
@@ -0,0 +1,29 @@
+import db from '../database/config/db.js';
+
+/**
+ * Get all user data by ID
+ * @param {number} userId - User ID
+ * @returns {Promise<Object>} - User data
+ */
+export async function getUserData(userId) {
+  const connection = await db.getConnection();
+  try {
+    const [rows] = await connection.query(
+      `SELECT 
+        u.id, 
```

Comment:
The id key for the user must be changed to `user_id` to match the database values.
```suggestion
        u.user_id,
```